docs/data_pack.md

```diff
@@ -1,5 +1,31 @@
 # DataPack Tutorial
 
+## Get primitive data
+`DataPack.get_data()` is commonly used to retrieve primitive data from a `DataPack`. User can request particular data fields within the range of a particular `Annotation` or `AudioAnnotation` type. User request particular data fields by setting `request` and the search range by setting `context_type`.
+
+In forte, each annotation has a range which includes begin and end of annotation-specific data of that particular annotation. For `Annotation` type, range means the begin index and end index of characters under `Annotation` type in the `text` payload of the `DataPack`. For `AudioAnnotation` type, range means the begin index and end index of sound sample under `AudioAnnotation` type in the `audio` payload of the `DataPack`. For an `Token` instance which is a subtype of `Annotation`, its annotation-specific data is `text` and therefore range means the begin and end of characters of that `Token` instance. For an `Recording` instance which is a subtype of `AudioAnnotation`, its annotation-specific data is `audio` and there range means the begin and end index of that `Recording` instance.
+
+
+For example, if User wants to get primitive data of `AudioAnnotation` from a `DataPack` instance `pack`. User can call the function like the code blow. It returns a generator that User can iterate over.
+`AudioAnnotation` is passed into the method as parameter `context_type`.
+```python
+pack.get_data(AudioAnnotation)
+```
+
+For example, if User wants to get primitive data of `AudioUtterance` from a `DataPack` instance `pack` and specific data fields such as `speaker` for `AudioUtterance` entry. User can call the function like the code blow.
+
+```python
+pack.get_data(AudioAnnotation,
+                {
+                AudioUtterance:
+                    {"fields": ["speaker"]}
+                }
+            )
+```
+
+Based on the idea of "range", in the example code, entry `AudioUtterance` will be searched in `DataPack.audio_annotations` and the requested data field `speaker` will be included in the generator's data.
+
```

Comment:
Didn't mention that this step is actually a generator. It would be easier to use a more concrete example like "Sentence" and "Token" to get people familiar with the concept, then extend the ide to "AudioAnnotation"